Layout variations

katzmo opened the pull request:
Implements https://github.com/moreonion/campaignion/pull/245

katzmo committed:
Activate “big banner” layout variation

katzmo committed:
Render the background image for the banner layout

torotil committed:
Let campaignion_layout handle rendering of the background image.

katzmo committed:
Add a wrapper around forms

alex-mo commented on template.php:
```diff
@@ -21,6 +21,22 @@
+/**
+ * Make theme layouts available.
+ */
+function campaignion_foundation_campaignion_layout_info() {
```
Isn't there a `Implements hook_campaignion_layout_info().` missing?

katzmo replied:
Maybe. Maybe not. I feel like [Drupal docstings](https://www.drupal.org/docs/develop/standards/api-documentation-and-comment-standards) are science of their own that I have not mastered yet.

torotil replied:
Yep, proper style would be to have `Implements …` as the first line in the comment.

katzmo replied:
Only for this hook? Or everything else in the file that is not a preprocess or custom function, too?

torotil replied:
@katzmo It’s the standard for all [hook implementations](https://www.drupal.org/docs/develop/standards/api-documentation-and-comment-standards#hookimpl).

alex-mo reviewed (commented):
Looks OK to me. Disclaimer: I am not at all familiar with layout variations.

katzmo committed:
Fix docstrings to adher to Drupal comment standards

alex-mo committed:
Move $form_fields to page preprocess function

This allows changing the set of form fields in derived themes and layout
variations.

katzmo commented on template.php:
```diff
@@ -86,6 +86,14 @@ function campaignion_foundation_process_node(&$vars) {
+/**
+ * Prepares variables for page templates.
+ */
+function campaignion_foundation_process_page(&$vars) {
```
Why `process_page` instead of `preprocess_page`?
